build/Dockerfile

```diff
@@ -14,36 +14,39 @@ RUN echo "4a7fa60f323ee1416a4b1425aefc37ea359e9d64df19c326a58953a97ad41ea5  go.t
 RUN tar -C /usr/local -xzf go.tar.gz
 ENV PATH=${PATH}:/usr/local/go/bin
 
+WORKDIR /build/windows-machine-config-operator/
+COPY .git .git
+
 # Build WMCB
-WORKDIR /build/windows-machine-config-bootstrapper/
+WORKDIR /build/windows-machine-config-operator/windows-machine-config-bootstrapper/
 COPY windows-machine-config-bootstrapper/ .
 RUN make build
 
 # Build hybrid-overlay
-WORKDIR /build/ovn-kubernetes/
+WORKDIR /build/windows-machine-config-operator/ovn-kubernetes/
 COPY ovn-kubernetes/ .
-WORKDIR /build/ovn-kubernetes/go-controller/
+WORKDIR /build/windows-machine-config-operator/ovn-kubernetes/go-controller/
 RUN make windows
 
 # Build promu utility tool, needed to build the windows_exporter.exe metrics binary
-WORKDIR /build/promu/
+WORKDIR /build/windows-machine-config-operator/promu/
 COPY promu/ .
 RUN make build
 
 # Build windows_exporter
-WORKDIR /build/windows_exporter/
+WORKDIR /build/windows-machine-config-operator/windows_exporter/
 COPY windows_exporter/ .
-RUN GOOS=windows /build/promu/promu build -v
+RUN GOOS=windows /build/windows-machine-config-operator/promu/prometheus-promu build -v
```

Comment:
I don't think changing the binary name is desirable - the `prometheus-promu` name is not for it upstream and as far as I can tell only comes from the openshift fork's downstream repo name (per the `org-name` schema used in these cases).

Looking at promu's Makefile, I think instead of running `make build` to build it, we might want something like `go install github.com/prometheus/promu`.

Reply:
`go install github.com/prometheus/promu` is going to be a problem in brew. I don't think we can `go install` anything in that environment. But we are already have that conundrum given the way promu is [built](https://github.com/prometheus/promu/blob/master/Makefile#L17):
```
build:
	@echo ">> installing promu"
	GO111MODULE=$(GO111MODULE) GOOS= GOARCH= $(GO) install github.com/prometheus/promu
	@echo ">> rebuilding binaries using ${PROMU} with ${PREFIX}"
	GO111MODULE=$(GO111MODULE) $(PROMU) build --prefix $(PREFIX)
```

Reply:
@PratikMahajan can't you just `COPY .git .git` in the kube binary building stage? At least for the moment.

Reply:
```
diff --git a/build/Dockerfile b/build/Dockerfile
index 5e84c7f..b1f79c2 100644
--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -15,7 +15,6 @@ RUN tar -C /usr/local -xzf go.tar.gz
 ENV PATH=${PATH}:/usr/local/go/bin
 
 WORKDIR /build/windows-machine-config-operator/
-COPY .git .git
 
 # Build WMCB
 WORKDIR /build/windows-machine-config-operator/windows-machine-config-bootstrapper/
@@ -36,9 +35,10 @@ RUN make build
 # Build windows_exporter
 WORKDIR /build/windows-machine-config-operator/windows_exporter/
 COPY windows_exporter/ .
-RUN GOOS=windows /build/windows-machine-config-operator/promu/prometheus-promu build -v
+RUN GOOS=windows /build/windows-machine-config-operator/promu/promu build -v
 
 # Build Kubernetes node binaries
+COPY .git /build/windows-machine-config-operator/.git
 WORKDIR /build/windows-machine-config-operator/kubernetes/
 COPY kubernetes/ .
 ENV KUBE_BUILD_PLATFORMS windows/amd64
```

Reply:
@aravindhp since the local go.mod file declares the local module context to be promu here, in this case:
`go install github.com/prometheus/promu` == `go install .` == `go build -o $GOROOTBIN/dir/promu.exe .`,
so it will not download anything - I think that part shouldn't be a problem in brew.

It seems promu is using a standard Prometheus Makefile, which will usually download promu and then build the target with promu.
That's a bit weird here, as it'll first build promu with `go install` and then rebuild promu with the promu that was just built.
The binary that is built in the second step will currently always get its name from the repo root dir name, which is why we're getting `prometheus-promu` here.
I've opened https://github.com/prometheus/promu/pull/197 so it'll always be `promu` to make that a bit more ergonomic for us.

I think the far bigger issue here is that `windows_exporter` does not include a vendor directory and will thus need to pull in its deps during the build, which probably won't work in brew. We may have to add the vendor dir to our fork.